application/controllers/QuestionAdministrationController.php

```diff
@@ -473,11 +473,20 @@ public function actionSaveQuestionData()
             }
 
             if ($question->survey->active == 'N') {
-                // Clean subquestions before save.
-                $question->deleteAllSubquestions();
                 // If question type has subquestions, save them.
                 if ($question->questionType->subquestions > 0) {
-                    $this->storeSubquestions(
+                    // Delete subquestions that do not appear in submit
```

Comment:
Why not order

1. deleteMissingSubquestions
2. updateSubquestions
3. addNewSubquestions

With updateSubquestions : checking if question exist (not only `new`) and same for addNewSubquestions ?

Reply:
Sorry, don't understand... what's the upside of the order change?

Reply:
after adding new subquestions : you have new qid related to this subqid.

Then you can not really check if question exist 

It's not a real issue , but seems more logical : deleted , update , add …